ethereum/contracts/bridge/L1WethBridge.sol

```diff
@@ -74,33 +92,51 @@ contract L1WethBridge is IL1Bridge, AllowListed, ReentrancyGuard {
     /// @notice _factoryDeps[1] == a raw bytecode of proxy that is used as L2 WETH bridge
     /// @param _l2WethAddress Pre-calculated address of L2 WETH token
     /// @param _governor Address which can change L2 WETH token implementation and upgrade the bridge
+    function initialize(
+        bytes[] calldata _factoryDeps,
+        address _l2WethAddress,
+        address _governor
+    ) external reentrancyGuardInitializer {
+        require(_l2WethAddress != address(0), "L2 WETH address cannot be zero");
+        require(_governor != address(0), "Governor address cannot be zero");
+        require(_factoryDeps.length == 2, "Invalid factory deps length provided");
+
+        l2WethAddress = _l2WethAddress;
+        l2Governor = _governor;
+
+        factoryDepsHash = keccak256(abi.encode(_factoryDeps));
+    }
+
+    /// @dev Initializes a contract bridge for later use. Expected to be used in the proxy
+    /// @dev During initialization deploys L2 WETH bridge counterpart as well as provides some factory deps for it
+    /// @param _factoryDeps A list of raw bytecodes that are needed for deployment of the L2 WETH bridge
+    /// @notice _factoryDeps[0] == a raw bytecode of L2 WETH bridge implementation. Note this deploys the Weth token
+    /// implementation and proxy upon initialization
+    /// @notice _factoryDeps[1] == a raw bytecode of proxy that is used as L2 WETH bridge
     /// @param _deployBridgeImplementationFee The fee that will be paid for the L1 -> L2 transaction for deploying L2
     /// bridge implementation
     /// @param _deployBridgeProxyFee The fee that will be paid for the L1 -> L2 transaction for deploying L2 bridge
     /// proxy
-    function initialize(
+    function initializeChain(
+        uint256 _chainId,
         bytes[] calldata _factoryDeps,
-        address _l2WethAddress,
-        address _governor,
         uint256 _deployBridgeImplementationFee,
         uint256 _deployBridgeProxyFee
-    ) external payable reentrancyGuardInitializer {
-        require(_l2WethAddress != address(0), "L2 WETH address cannot be zero");
-        require(_governor != address(0), "Governor address cannot be zero");
-        require(_factoryDeps.length == 2, "Invalid factory deps length provided");
+    ) external payable {
```

Comment:
Anyone can call this function and reinitialize the contract. What prevents from this?

Reply:
Yes, thank you corrected.